File: ATS_BE/src/main/java/com/ats/auth/service/AuthService.java
package com.ats.auth.service;

import com.ats.auth.dto.LoginRequest;
import com.ats.auth.dto.LoginResponse;
import com.ats.auth.dto.RefreshTokenRequest;
import com.ats.auth.dto.SsoLoginRequest;
import com.ats.auth.dto.UserProfileResponse;

public interface AuthService {
    LoginResponse login(LoginRequest request);
    LoginResponse ssoLogin(SsoLoginRequest request);
    LoginResponse refreshToken(RefreshTokenRequest request);
    void logout(String token);
    UserProfileResponse getCurrentUser();
}

File: ATS_FE/src/features/auth/services/auth.service.ts
import { apiClient } from '@/shared/services/api.client';
import type { ApiResponse } from '@/shared/types/api.types';
import type {
  LoginRequest,
  SsoLoginRequest,
  RefreshTokenRequest,
  LoginResponse,
  UserProfileResponse,
} from '../types/auth.types';

/**
 * Authentication & Authorization API service
 */
export const authService = {
  /**
   * User login with credentials
   */
  login: async (data: LoginRequest) => {
    return apiClient.post<ApiResponse<LoginResponse>>('/auth/login', data);
  },

  /**
   * SSO/LDAP federated login callback
   */
  ssoLogin: async (data: SsoLoginRequest) => {
    return apiClient.post<ApiResponse<LoginResponse>>('/auth/login/sso', data);
  },

  /**
   * Refresh expired JWT access token
   */
  refreshToken: async (data: RefreshTokenRequest) => {
    return apiClient.post<ApiResponse<LoginResponse>>('/auth/refresh', data);
  },

  /**
   * Invalidate current JWT token session
   */
  logout: async () => {
    return apiClient.post<ApiResponse<null>>('/auth/logout');
  },

  /**
   * Get current authenticated user profile
   */
  getCurrentUser: async () => {
    return apiClient.get<ApiResponse<UserProfileResponse>>('/auth/me');
  },
};
